### PR TITLE
Update sub component names

### LINKED ISSUE #1075: Update compound components with correct sub-components name
Result of discussion #864 

Update our compound components with correct naming-scheme for sub-components. 

* Do not repeat "parent" name in sub-components
* Sub-component should have a clear name, do we need `Card.HeaderTitle` in `Car.dHeader`? 

Renaming needs to be done next to existing names, and make list of "old" names to be flagged as deprecated in later release. 

Example of interim supporting old and new:

```typescript
import { Card as CardWrapper, CardProps } from './Card'
import { CardActions } from './CardActions'
import { CardMedia } from './CardMedia'
import { CardHeader } from './CardHeader'
import { CardHeaderTitle } from './CardHeaderTitle'

type CardCompoundProps = typeof CardWrapper & {
  // Old
  CardActions: typeof CardActions
  CardHeader: typeof CardHeader
  CardMedia: typeof CardMedia
  CardHeaderTitle: typeof CardHeaderTitle
  // New
  Actions: typeof CardActions
  Header: typeof CardHeader
  Media: typeof CardMedia
  HeaderTitle: typeof CardHeaderTitle
}

const Card = CardWrapper as CardCompoundProps

Card.CardActions = CardActions
Card.CardHeader = CardHeader
Card.CardMedia = CardMedia
Card.CardHeaderTitle = CardHeaderTitle

Card.Actions = CardActions
Card.Header = CardHeader
Card.Media = CardMedia
Card.HeaderTitle = CardHeaderTitle

export { Card }
export type { CardProps }


```

### PR DESCRIPTION
**Pull request by wenche**
Resolve #1075

### CODE REVIEW

**Comment by wenche** [comment]
I don't think we need to change `<Breadcrumbs>` and `<Breadcrumb>`

**Comment by wenche** [comment]
Ended up changing the following components:

- Accordion
- Card
- Banner
- List
- Menu
- Popover
- Snackbar
- Tabs

The rest of the compound components were already following this naming format.

File: libraries/core-react/src/components/Tabs/index.ts [hunk 1]
```diff
@@ -7,16 +7,24 @@ import { TabPanel } from './TabPanel'
...
+  // New
+  Panels: typeof TabPanels
+  Panel: typeof TabPanel
 }
 
 const Tabs = BaseComponent as TabsCompoundProps
 
 Tabs.Tab = Tab
+// Deprecated
 Tabs.TabList = TabList
```
Comment (mimarz) [inline libraries/core-react/src/components/Tabs/index.ts:22]: We skip `Tab.TabList`  -> `Tab.List` ?

File: libraries/core-react/stories/components/Banner.stories.tsx [hunk 2]
```diff
@@ -11,105 +11,113 @@ const icons = {
   mood_sad,
 }
 
-const { BannerIcon, BannerMessage, BannerActions } = Banner
-Icon.add(icons)
+const { Icon, Message, Actions } = Banner
```
Comment (mimarz) [inline libraries/core-react/stories/components/Banner.stories.tsx:14]: Remove this desconstruct and remove rename of  `Icon` -> `EdsIcon`?

File: libraries/core-react/stories/components/Lists.stories.tsx [hunk 3]
```diff
@@ -2,76 +2,85 @@ import React from 'react'
 import { List, ListProps } from '@components'
 import { Meta, Story } from '@storybook/react'
 
-const { ListItem } = List
+const { Item } = List
```
Comment (mimarz) [inline libraries/core-react/stories/components/Lists.stories.tsx:5]: No longer needed i presume?

File: libraries/core-react/stories/components/Menu.stories.tsx [hunk 4]
```diff
@@ -34,11 +34,25 @@ Icon.add({
   cable,
 })
 
-const { MenuItem, MenuSection } = Menu
+const { Item, Section } = Menu
 const { Actions, Header } = TopBar
```
Comment (mimarz) [inline libraries/core-react/stories/components/Menu.stories.tsx:38]: Maybe change `Topbar` components to follow same dot annotation we talked about and remove deconstruct?

File: libraries/core-react/stories/components/Popover.stories.tsx [hunk 5]
```diff
@@ -3,8 +3,8 @@ import styled from 'styled-components'
 import { Typography, Button, Popover, PopoverProps, Card } from '@components'
 import { Meta, Story } from '@storybook/react'
 
-const { PopoverAnchor, PopoverTitle, PopoverContent } = Popover
-const { CardActions } = Card
+const { Anchor, Title, Content } = Popover
+const { Actions } = Card
```
Comment (mimarz) [inline libraries/core-react/stories/components/Popover.stories.tsx:7]: No longer needed?

File: libraries/core-react/stories/components/Snackbar.stories.tsx [hunk 6]
```diff
@@ -2,12 +2,20 @@ import React, { useState } from 'react'
 import { Snackbar, SnackbarProps, Button } from '@components'
 import { Story, Meta } from '@storybook/react'
 
-const { SnackbarAction } = Snackbar
+const { Action } = Snackbar
```
Comment (mimarz) [inline libraries/core-react/stories/components/Snackbar.stories.tsx:5]: 🔥

File: libraries/core-react/stories/components/Table.stories.tsx [hunk 7]
```diff
@@ -18,6 +18,28 @@ Icon.add({ chevron_down, chevron_up })
 
 const { Caption, Body, Row, Cell, Head } = Table
```
Comment (mimarz) [inline libraries/core-react/stories/components/Table.stories.tsx:19]: Maybe make a seperate issue for converting existing stories to use dot-style in examples 🤔

Comment (wenche) [reply libraries/core-react/stories/components/Table.stories.tsx:19]: No, I think we should do it now, doesnt take long. Just didn't think about it 🙃

File: libraries/core-react/src/components/Tabs/index.ts (hunk 1 above)
Comment (wenche) [reply libraries/core-react/src/components/Tabs/index.ts:22]: This one was hard 🙈  I did change it initially, but when I updated the storybook example even I was confused by the name List inside a Tabs 🤪  So I changed it back and added a comment about this in the commit message.

File: libraries/core-react/stories/components/Banner.stories.tsx (hunk 2 above)
Comment (wenche) [reply libraries/core-react/stories/components/Banner.stories.tsx:14]: Some syntax fiddling gives this as an option, I suggest we use that approach 🤷‍♀️ 

```
subcomponents: {
    Message: Banner.Message,
    Icon: Banner.Icon,
    Actions: Banner.Actions,
  },
```

File: libraries/core-react/stories/components/Lists.stories.tsx (hunk 3 above)
Comment (wenche) [reply libraries/core-react/stories/components/Lists.stories.tsx]: It was needed for the default export as `Meta` but changed the syntax slightly

File: libraries/core-react/stories/components/Menu.stories.tsx (hunk 4 above)
Comment (wenche) [reply libraries/core-react/stories/components/Menu.stories.tsx]: Yes, updated all examples even those who was not actually changed. 👍

File: libraries/core-react/stories/components/Popover.stories.tsx (hunk 5 above)
Comment (wenche) [reply libraries/core-react/stories/components/Popover.stories.tsx]: Same as before

**Review by mimarz — APPROVED** [review]
LGTM 👍

Great you added the description to the components 🤩

**Pull request merged**